adds hover state to time travel buttons

dabutvin opened the pull request:
also updated the active state for these to match the hover state for consistency

![hover](https://user-images.githubusercontent.com/5168796/43155045-2603e9de-8f2b-11e8-8e6f-6c0ad4540cfa.gif)

dvdsgl commented:
Oh that is not pretty. Can you try a more subtle effect? Also make the cursor a pointer on hover.

dabutvin commented:
made some updates. Let me know what you think :)

 - added border, box-shadow, and vertical shift on hover
 - removed shadow and shift on click
 - made the cursor pointer on hover
 - made the default outline focus for keyboard navigation only


![2018-07-24 21 07 03](https://user-images.githubusercontent.com/5168796/43179231-8536729a-8f86-11e8-8ee4-537e1bc55b7a.gif)

dvdsgl commented:
Nice improvements.

Can you remove the vertical displacements? It's causing the scrollbar to flicker on click for me.

dabutvin commented:
@dvdsgl I updated to use `transform: translateY(-1px);` instead of `top: -1px` on hover

Can you see if this fixes the scrollbar issue on your setup?

If it works, I think keeping the rise on hover would be valuable to give the sense that the button is being picked up.

dvdsgl commented on docs/time-travel/index.html:
```diff
@@ -30,10 +30,10 @@ <h2 class="history__count">
+        <button id="js-button-first"><span tabindex="-1">First</span></button>
+        <button id="js-button-previous"><span tabindex="-1">Previous</span></button>
+        <button id="js-button-next"><span tabindex="-1">Next</span></button>
+        <button id="js-button-last"><span tabindex="-1">Latest</span></button>
```
What's with the `tabindex="-1"`?

Otherwise, this looks good to merge.

dabutvin replied:
This is to skip these child elements when tabbing through the buttons by keyboard.

I made it so you still get the blue focus outline when keyboarding, but not when clicking because when you click the button, the child span gets focus.

dvdsgl replied:
Sorry, why would keyboard users want to skip these elements?

dabutvin replied:
We are only skipping the span elements within the button. You can still tab to the button elements.

This allows us to add a focus style that does not include the native big blue outline, but retain the big blue outline for those that are navigating by keyboard. 

The native focus outline is really handy for those that rely on it as their sole means of navigation.

Here is a really good explanation of this technique.
https://stackoverflow.com/a/47401699/2019445

I believe there are newer ways of handling this situation, but this is my favorite for it's simplicity.

dvdsgl commented:
Nice work!
